src/backend/src/yaksur_pup.c

```diff
@@ -10,12 +10,13 @@
 #include "yaksuri.h"
 
 static int ipup(const void *inbuf, void *outbuf, uintptr_t count, yaksi_type_s * type,
-                yaksi_info_s * info, yaksa_op_t op, yaksi_request_s * request)
+                yaksi_info_s * info, yaksa_op_t op, yaksi_request_s * request,
+                bool cache_gpudriver_id)
 {
     int rc = YAKSA_SUCCESS;
     yaksuri_request_s *reqpriv = (yaksuri_request_s *) request->backend.priv;
 
-    if (reqpriv->gpudriver_id != YAKSURI_GPUDRIVER_ID__UNSET)
+    if (cache_gpudriver_id && reqpriv->gpudriver_id != YAKSURI_GPUDRIVER_ID__UNSET)
```

Comment:
`cache_gpudirver_id` seems to be a bit misleading, since `reqpriv->gpudriver_id` is still set and used. The direct intention is to "always_query_attr", right?

Reply:
Yes, you are right. I changed this to `always_query_ptr_attr`.